cmd/util/cmd/diff-states/cmd.go

```diff
@@ -0,0 +1,528 @@
+			stateCommitment := parseStateCommitment(flagStateCommitment2)
+			payloads2, err = readTrie(flagState2, stateCommitment)
+		}
+		return
+	})
+
+	err := group.Wait()
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+
+	log.Info().Msg("Finished loading payloads")
+
+	return
+}
+
+func payloadsToRegisters(payloads1, payloads2 []*ledger.Payload) (registers1, registers2 *registers.ByAccount) {
+
+	log.Info().Msg("Creating registers from payloads")
+
+	var group errgroup.Group
+
+	group.Go(func() (err error) {
+		log.Info().Msgf("Creating registers from first payloads (%d)", len(payloads1))
+
+		registers1, err = registers.NewByAccountFromPayloads(payloads1)
+
+		log.Info().Msgf(
+			"Created %d registers from payloads (%d accounts)",
+			registers1.Count(),
+			registers1.AccountCount(),
+		)
+
+		return
+	})
+
+	group.Go(func() (err error) {
+		log.Info().Msgf("Creating registers from second payloads (%d)", len(payloads2))
+
+		registers2, err = registers.NewByAccountFromPayloads(payloads2)
+
+		log.Info().Msgf(
+			"Created %d registers from payloads (%d accounts)",
+			registers2.Count(),
+			registers2.AccountCount(),
+		)
+
+		return
+	})
+
+	err := group.Wait()
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+
+	log.Info().Msg("Finished creating registers from payloads")
+
+	return
+}
+
+var accountsDiffer = errors.New("accounts differ")
+
+func diff(
+	registers1 *registers.ByAccount,
+	registers2 *registers.ByAccount,
+	chainID flow.ChainID,
+	rw reporters.ReportWriter,
+) {
+	log.Info().Msg("Diffing accounts")
+
+	err := registers1.ForEachAccount(func(accountRegisters1 *registers.AccountRegisters) (err error) {
+		owner := accountRegisters1.Owner()
+
+		if !registers2.HasAccountOwner(owner) {
+			rw.Write(accountMissing{
+				Owner: owner,
+				State: 2,
+			})
+
+			return nil
+		}
+
+		accountRegisters2 := registers2.AccountRegisters(owner)
+
+		if accountRegisters1.Count() != accountRegisters2.Count() {
+			rw.Write(countDiff{
+				Owner:  owner,
+				State1: accountRegisters1.Count(),
+				State2: accountRegisters2.Count(),
+			})
+		}
+
+		err = accountRegisters1.ForEach(func(owner, key string, value1 []byte) error {
+			var value2 []byte
+			value2, err = accountRegisters2.Get(owner, key)
+			if err != nil {
+				return err
+			}
+
+			if !bytes.Equal(value1, value2) {
+
+				if flagRaw {
+					rw.Write(rawDiff{
+						Owner:  owner,
+						Key:    key,
+						Value1: value1,
+						Value2: value2,
+					})
+				} else {
+					// stop on first difference in accounts
+					return accountsDiffer
+				}
+			}
+
+			return nil
+		})
+		if err != nil {
+			if flagRaw || !errors.Is(err, accountsDiffer) {
+				return err
+			}
+
+			address, err := common.BytesToAddress([]byte(owner))
+			if err != nil {
+				return err
+			}
+
+			migrations.NewCadenceValueDiffReporter(
+				address,
+				chainID,
+				rw,
+				true,
+				flagNWorker,
+			).DiffStates(
+				accountRegisters1,
+				accountRegisters2,
+				migrations.AllStorageMapDomains,
+			)
```

Comment:
> Yeah, this can be maybe added, but not quite sure how

One approach is to compare non-atree registers when diffing in Cadence-level, and log difference in `rawDiff` message.

Reply:
For now the assumption of the command is that if states are compared on a value level, then only values are compared